Classes/Service/GeocodeService.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace FriendsOfTYPO3\TtAddress\Service;
+
+/**
+ * This file is part of the "tt_address" Extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+use FriendsOfTYPO3\TtAddress\Domain\Model\Dto\Settings;
+use TYPO3\CMS\Core\Cache\CacheManager;
+use TYPO3\CMS\Core\Cache\Frontend\FrontendInterface;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Database\Query\QueryHelper;
+use TYPO3\CMS\Core\Database\Query\Restriction\DeletedRestriction;
+use TYPO3\CMS\Core\SingletonInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Service for category related stuff
+ *
+ * thanks to https://github.com/b13/t3ext-geocoding for inspiration
+ */
+class GeocodeService implements SingletonInterface
+{
+
+    /** @var Settings */
+    protected $extensionSettings;
+
+    /** @var int */
+    protected $cacheTime = 7776000;
+
+    /** @var string  */
+    protected $geocodingUrl = 'https://maps.googleapis.com/maps/api/geocode/json?language=de&sensor=false';
+
+    public function __construct()
+    {
+        $this->extensionSettings = GeneralUtility::makeInstance(Settings::class);
+        $this->geocodingUrl .= '&key=' . $this->extensionSettings->getGoogleMapsKeyGeocoding();
```

Comment:
no need to use $this->extensionSettings as a property, if it is only used for one parameter.

Reply:
I will switch to a more simple way and require the CLI as argument